### PR TITLE
Updated regularization

### PR DESCRIPTION
**Pull request by gabehope**
Changed the way regularization works for the regression model. Now the regularization is just a weighted L2 (can also be set to L1) regularizer that scales with lambda. 

#### Other changes
* Added support for per-timestep labels in rungrid (untested)
* Added flags for setting regularization, prediction loss and window to rungrid
* Updated my grid scripts to use your much nicer format
* 1 or 2 very minor fixes

### CODE REVIEW

File: src/predict.py [hunk 1]
```diff
@@ -27,11 +29,14 @@ def from_tf_variables(self, eta_tf, sess):
...
+    def reg_eta(self, eta):
+        if isinstance(eta, np.ndarray):
+            return self.reg_weight * np.sum(np.abs(eta) ** self.reg_power)
+        else:
+            return self.reg_weight * np.sum(np.abs(eta[0]) ** self.reg_power)
 
-    def log_p_eta_tf(self, eta):
-        raise NotImplementedError()
+    def reg_eta_tf(self, eta):
+        return self.reg_weight * tf.reduce_sum(tf.abs(eta[0]) ** self.reg_power)
```
Comment (michaelchughes) [inline src/predict.py]: When would we need eta[0] instead of just eta? Just curious...

**Comment by gabehope** [comment]
The tensor flow variables for eta are always passed around in a list so that it can easily support more than 1 trainable variable (like the window weighting for per-timestep predictions).

**Comment by gabehope** [comment]
Also, thanks for approving!
